controller/src/main/java/org/jboss/as/controller/registry/OperationTransformerRegistry.java

```diff
@@ -355,7 +355,10 @@ private OperationTransformerRegistry createChild(Iterator<PathElement> iterator,
         }
 
         private void registerTransformer(Iterator<PathElement> iterator, String value, String operationName,  OperationTransformerEntry entry) {
-            get(value).registerTransformer(iterator, operationName, entry);
+            OperationTransformerRegistry operationTransformerRegistry = get(value);
+            if (operationTransformerRegistry != null) {
+                operationTransformerRegistry.registerTransformer(iterator, operationName, entry);
+            }
```

Comment:
I've reviewed the changes but I don't feel capable to approve them as they are. In general, they look harmless; we can argue that if the CI is passing, then we are fine, but there are possibilities for introducing an unexpected bug or a side effect.

I think we should be more cautious with those changes.

Specifically, this one, if `operationTransformerRegistry` is null the old behavior was to get a null pointer exception, but now the method is silently invoked without registering the expected transformer and that can be a wrong result difficult to track and debug on a mixed domain scenario. Maybe it could be better to keep the NPE or throw an IAE if it is considered an invalid state. This argument can be also extrapolated to [YamlConfigurationExtension.java](https://github.com/wildfly/wildfly-core/pull/5012/files#diff-d50b9b2a6b6cbf891246ead8a48357b7fdd24ef0ba76f35bda6e992afb26c742), [OperationValidator.java](https://github.com/wildfly/wildfly-core/pull/5012/files#diff-ef60c464eae6dddb2e6505f8c47b941da01873c707d5ab547349340abc40b647), in which case leaving the code as it is now is also a valid alternative.